fontforge/namelist.c

```diff
@@ -200,9 +200,11 @@ const char *StdGlyphName(char *buffer, int uni,enum uni_interp interp,NameList *
 			(name = nl->unicode[up][ub][uc])!=NULL )
 	    break;
 	}
+    } else {
+	LogError( _("Warning: StdGlyphName returning name for value %d outside of Unicode range\n"), uni );
     }
```

Comment:
One of the problems with python/native scripting is this Logerror. It stops abruptly in commandline mode, which is ok, but in GUI mode it gives you a message and keeps on trucking as if nothing wrong.
Ideally, you want it to stop too. So you will want to tackle this with two plans. For nameslist, the ideal situation is keep your house clean, and if you have an error, return an error when you fail. You want to push the problem upstream and because LogError has a dual personality, you may want to push this function into python.c and scripting.c instead. Short term, commandline will exit on LogError(), long term strategy, we fix LogError() to not exit, and we do the proper stop at the root level where you process instructions, not deep in the python commands.
If you do a diff between 20120731 scripting.c and current scripting.c you'll probably see a lot of editing in trying to push this exit() out into the  command processing routine in anticipation of fixing LogError(). This should fix the different behaviour between commandline scripting stopping like it should, vs GUI scripting doing let's carry-on as if nothing really happened.

Reply:
I actually don't want it to stop in either case. I just want the error to be evident to a careful observer.

Reply:
I think some of my grousing may have been lost in the shuffle. I've added a new function `StdGlyphNameBoundsCheck()`. That's the one with the error-handling conventions. This one doesn't have error handling *conventions*, but I do want it to print a warning (and not stop) when it's called with a weird value. 

My goal with this change is to 1) add a function that has error handling conventions and is correct, and use that when calling from python, 2) fix the crash and add the warning printing in the existing function. 

If someone else (e.g. @JoesCat) wants to come in after and remove all the calls to the existing function by adapting the callers to have error handling, that would be great.

Reply:
I don't see in `nouiutil.c` where `LogError` would exit, but I guess I can test that easily enough.